### PR TITLE
[batch] save pod status

### PR DESCRIPTION
**Pull request by jigold**
@danking Feel free to reassign if you have too much to do.

Requires a database change. I realize there's a lot of code duplication here with how we get/show the logs. It might be worth generalizing this in the future if there's a third thing that follows this pattern.

### CODE REVIEW

File: batch/batch/batch.py [hunk 1]
```diff
@@ -259,7 +259,32 @@ class Job:
...
         return None
 
-    async def _mark_job_task_complete(self, task_name, log, exit_code, new_state):
+    async def _read_pod_statuses(self):
+        async def _read_pod_status(jt):
+            pod_status = await db.jobs.get_pod_status(*self.id, jt.name)
+            return jt.name, pod_status
+
+        future_pod_statuses = asyncio.gather(*[_read_pod_status(jt) for idx, jt in enumerate(self._tasks)
+                                               if idx < self._task_idx])
```
Comment (danking) [inline batch/batch/batch.py]: We should make one call to the database. If I'm reading this right we make as many as three calls to the database, each one looking up the same record, but asking for a different column. Just grab all three and take the non-null ones

File: batch/batch/batch.py [hunk 2]
```diff
@@ -640,6 +667,17 @@ def to_dict(self):
     abort(404)
 
 
+async def _get_pod_status(batch_id, job_id, user):
+    job = await Job.from_db(batch_id, job_id, user)
+    if not job:
+        abort(404)
+
+    pod_statuses = await job._read_pod_statuses()
```
Comment (danking) [inline batch/batch/batch.py]: We should get the pod statuses when we get the job the first time, why make a second trip?

Comment (jigold) [reply batch/batch/batch.py]: I think I addressed this. It still requires an async call to get the pod status for the current task but i use the already fetched ones for previous tasks.

**Comment by jigold** [comment]
Should we be transferring the data for the pod status for each call to the db or only if needed? I don't think they'll get too big, but it's something to consider -- tradeoff between db calls and transferring data when not needed.

**Comment by jigold** [comment]
See #6490

**Pull request merged**